test/TestMatchError.py

```diff
@@ -0,0 +1,29 @@
+from ansiblelint.errors import MatchError
+from ansiblelint.rules.AlwaysRunRule import AlwaysRunRule
+from ansiblelint.rules.BecomeUserWithoutBecomeRule import BecomeUserWithoutBecomeRule
+
+import pytest
+
+
+def test_matcherror_compare():
+    assert MatchError("foo") == MatchError("foo")
+
+
+@pytest.mark.parametrize(
+    ('a', 'b'), (
+        # sorting by message
+        (MatchError("z"), MatchError("a")),
+        # filenames takes priority in sorting
+        (MatchError("a", filename="b"), MatchError("a", filename="a")),
+        # rule id 501 > rule id 101
+        (MatchError(rule=BecomeUserWithoutBecomeRule), MatchError(rule=AlwaysRunRule))
+    ))
+def test_matcherror_sort(a, b):
+    assert b < a
+    assert a > b
+    assert a != b
```

Comment:
A test must contain just one assert. This will ensure that the tests show up separately and early failures don't prevent other asserts from running.

Reply:
I do not think that it makes sense to split this into multiple tests, especially as it is already parametrized.

Doing a ``b < a and a < b and a != b`` would also use a single assert but it would clearly not produce better errors if the test fails.

Can you please provide some links that sustain the use of multiple asserts is inherently wrong? IMHO, that is one case where they are ok. 

I doubt anyone pays us by number of tests written, so no need to boost their number ;)

Reply:
I agree with @ssbarnea here, I don't think it makes sense to split this into several tests just to keep a 'one assert per test' policy.

Reply:
I found a way to fix it!

Reply:
Well, the idea behind the "single assert" is not that only one keyword is allowed syntactically but that one single behavior should be tested and there should be one reason for the test to fail.

https://dev.to/donut87/one-assert-per-test---what-is-that-about-4l75